app/models/service_template.rb

```diff
@@ -75,6 +88,12 @@ def self.create_catalog_item(options, auth_user)
     end
   end
 
+  def self.class_from_request_data(data)
+    request_type = data['prov_type']
+    model_symbol = PROV_TYPE_TO_MODEL[request_type] || raise(ArgumentError, 'Invalid prov_type')
```

Comment:
We can/should avoid lookup from a hash and prefer naming convention. If the type starts with `generic_`, we can do an auto conversion from `generic_abc` to `ServiceTemplateAbc`; for all the rest, directly `ServiceTemplate`.